# **Automatidata project**
**Course 2 - Get Started with Python**

Welcome to the Automatidata Project!

You have just started as a data professional in a fictional data consulting firm, Automatidata. Their client, the New York City Taxi and Limousine Commission (New York City TLC), has hired the Automatidata team for its reputation in helping their clients develop data-based solutions.

The team is still in the early stages of the project. Previously, you were asked to complete a project proposal by your supervisor, DeShawn Washington. You have received notice that your project proposal has been approved and that New York City TLC has given the Automatidata team access to their data. To get clear insights, New York TLC's data must be analyzed, key variables identified, and the dataset ensured it is ready for analysis.

A notebook was structured and prepared to help you in this project. Please complete the following questions.

# Course 2 End-of-course project: Inspect and analyze data

In this activity, you will examine data provided and prepare it for analysis.  This activity will help ensure the information is,

1.   Ready to answer questions and yield insights

2.   Ready for visualizations

3.   Ready for future hypothesis testing and statistical methods
<br/>    

**The purpose** of this project is to investigate and understand the data provided.
  
**The goal** is to use a dataframe contructed within Python, perform a cursory inspection of the provided dataset, and inform team members of your findings. 
<br/>  
*This activity has three parts:*

**Part 1:** Understand the situation 
* Prepare to understand and organize the provided taxi cab dataset and information.

**Part 2:** Understand the data

* Create a pandas dataframe for data learning, future exploratory data analysis (EDA), and statistical activities.

* Compile summary information about the data to inform next steps.

**Part 3:** Understand the variables

* Use insights from your examination of the summary data to guide deeper investigation into specific variables.


<br/> 
Follow the instructions and answer the following questions to complete the activity. Then, you will complete an Executive Summary using the questions listed on the PACE Strategy Document.

Be sure to complete this activity before moving on. The next course item will provide you with a completed exemplar to compare to your own work. 



# **Identify data types and relevant variables using Python**


<img src="images/Pace.png" width="100" height="100" align=left>

# **PACE stages**


Throughout these project notebooks, you'll see references to the problem-solving framework PACE. The following notebook components are labeled with the respective PACE stage: Plan, Analyze, Construct, and Execute.

<img src="images/Plan.png" width="100" height="100" align=left>


## PACE: **Plan**

Consider the questions in your PACE Strategy Document and those below to craft your response:

### **Task 1. Understand the situation**

*   How can you best prepare to understand and organize the provided taxi cab information? 

Import the data with necessary packages and view the dataset. Identify the variables and datatypes present and work to understand their meaning. Clean the dataset to ensure it is ready for analysis.

<img src="images/Analyze.png" width="100" height="100" align=left>

## PACE: **Analyze**

Consider the questions in your PACE Strategy Document to reflect on the Analyze stage.

### **Task 2a. Build dataframe**
















Create a pandas dataframe for data learning, and future exploratory data analysis (EDA) and statistical activities.

**Code the following,**

*   import pandas as `pd`. pandas is used for buidling dataframes.

*   import numpy as `np`. numpy is imported with pandas

*   `df = pd.read_csv('Datasets\NYC taxi data.csv')`

**Note:** pair the data object name `df` with pandas functions to manipulate data, such as `df.groupby()`.

**Note:** As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

In [2]:
#Import libraries and packages listed above
import pandas as pd
import numpy as np

# Load dataset into dataframe
df = pd.read_csv('2017_Yellow_Taxi_Trip_Data.csv')
print("done")

done


### **Task 2b. Understand the data - Inspect the data**

View and inspect summary information about the dataframe by coding the following:

1. `df.head(10)`
2. `df.info()`
3. `df.describe()`

Consider the following two questions:

**Question 1:** When reviewing the `df.info()` output, what do you notice about the different variables? Are there any null values? Are all of the variables numeric? Does anything else stand out?

**Question 2:** When reviewing the `df.describe()` output, what do you notice about the distributions of each variable? Are there any questionable values?

Question 1: There are no null values in the dataset. 8 of the variables are float, 7 are integer, and 3 are objects (tpep_pickup_datetime, tpep_dropoff_datetime, store_and_fwd_flag). tpep_pickup_datetime and tpep_dropoff_datetime variables are date-time groups in the format "MM/DD/YYYY HH:MM:SS AM/PM". Dollar amount variables are float instead of currency. store_and_fwd_flag variable appears to be boolean (Y or N).The Unnamed: 0 variable appears to be unique ids for each trip in the dataset. Unclear what unit of measure is indicated by trip_distance (assuming miles).

Question 2: I believe the summary statistics given for Unnamed: 0, VendorID, RatecodeID, PULocationID, DOLocationID, and payment_type variables are largely not meaningful because they represent qualitative and not quantitative values. passenger_count has a min value of 0, which, on the surface, doesn't make sense - why would a trip be recorded if no passengers are present? It appears that in a majority of the trips, there is only one passenger. Similar to the min of passenger_count, trip_distance has a min value of 0. On the surface this doesn't seem to make sense - why would a trip be recorded if no travel occurred? fare_amount has a min value of -120: what does that mean? Is this how refunds are recorded? Did the driver/TLC owe money on that/those trip(s) somehow? fare_amount mean is approx. 13.03 whereas the value at 50% is 9.50, which indicates the distribution is skewed right - there are some outliers on the right side of this distribution skewing the mean. extra's mean is 0.33 whereas the value at 50% is 0.00, meaning this distribution is also skewed right. mta_tax is consistently 0.5 across all values in the dataset - it appears there may be tax credits present, where TLC gained 0.5 in some transactions. tip_amount is slightly skewed right with a mean of 1.84 and a value at 50% of 1.35. It appears that most rides do not incur tolls. improvement_surcharge is consistently 0.3 across all values in the dataset - there appears to be some -0.3 values in the dataset, which could be part of overall trip refunds. total_amount is skewed right with a mean of 16.31 and a value at 50% of 11.8, which makes sense since several of the component amount distributions are skewed right.

In [2]:
df.head(10)

Unnamed: 0  VendorID    tpep_pickup_datetime   tpep_dropoff_datetime  \
0    24870114         2   03/25/2017 8:55:43 AM   03/25/2017 9:09:47 AM   
1    35634249         1   04/11/2017 2:53:28 PM   04/11/2017 3:19:58 PM   
2   106203690         1   12/15/2017 7:26:56 AM   12/15/2017 7:34:08 AM   
3    38942136         2   05/07/2017 1:17:59 PM   05/07/2017 1:48:14 PM   
4    30841670         2  04/15/2017 11:32:20 PM  04/15/2017 11:49:03 PM   
5    23345809         2   03/25/2017 8:34:11 PM   03/25/2017 8:42:11 PM   
6    37660487         2   05/03/2017 7:04:09 PM   05/03/2017 8:03:47 PM   
7    69059411         2   08/15/2017 5:41:06 PM   08/15/2017 6:03:05 PM   
8     8433159         2   02/04/2017 4:17:07 PM   02/04/2017 4:29:14 PM   
9    95294817         1   11/10/2017 3:20:29 PM   11/10/2017 3:40:55 PM   

   passenger_count  trip_distance  RatecodeID store_and_fwd_flag  \
0                6           3.34           1                  N   
1                1           1.80           1                  N   
2                1           1.00           1                  N   
3                1           3.70           1                  N   
4                1           4.37           1                  N   
5                6           2.30           1                  N   
6                1          12.83           1                  N   
7                1           2.98           1                  N   
8                1           1.20           1                  N   
9                1           1.60           1                  N   

   PULocationID  DOLocationID  payment_type  fare_amount  extra  mta_tax  \
0           100           231             1         13.0    0.0      0.5   
1           186            43             1         16.0    0.0      0.5   
2           262           236             1          6.5    0.0      0.5   
3           188            97             1         20.5    0.0      0.5   
4             4           112             2         16.5    0.5      0.5   
5           161           236             1          9.0    0.5      0.5   
6            79           241             1         47.5    1.0      0.5   
7           237           114             1         16.0    1.0      0.5   
8           234           249             2          9.0    0.0      0.5   
9           239           237             1         13.0    0.0      0.5   

   tip_amount  tolls_amount  improvement_surcharge  total_amount  
0        2.76           0.0                    0.3         16.56  
1        4.00           0.0                    0.3         20.80  
2        1.45           0.0                    0.3          8.75  
3        6.39           0.0                    0.3         27.69  
4        0.00           0.0                    0.3         17.80  
5        2.06           0.0                    0.3         12.36  
6        9.86           0.0                    0.3         59.16  
7        1.78           0.0                    0.3         19.58  
8        0.00           0.0                    0.3          9.80  
9        2.75           0.0                    0.3         16.55

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22699 entries, 0 to 22698
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Unnamed: 0             22699 non-null  int64  
 1   VendorID               22699 non-null  int64  
 2   tpep_pickup_datetime   22699 non-null  object 
 3   tpep_dropoff_datetime  22699 non-null  object 
 4   passenger_count        22699 non-null  int64  
 5   trip_distance          22699 non-null  float64
 6   RatecodeID             22699 non-null  int64  
 7   store_and_fwd_flag     22699 non-null  object 
 8   PULocationID           22699 non-null  int64  
 9   DOLocationID           22699 non-null  int64  
 10  payment_type           22699 non-null  int64  
 11  fare_amount            22699 non-null  float64
 12  extra                  22699 non-null  float64
 13  mta_tax                22699 non-null  float64
 14  tip_amount             22699 non-null  float64
 15  to

In [4]:
df.describe()

Unnamed: 0      VendorID  passenger_count  trip_distance  \
count  2.269900e+04  22699.000000     22699.000000   22699.000000   
mean   5.675849e+07      1.556236         1.642319       2.913313   
std    3.274493e+07      0.496838         1.285231       3.653171   
min    1.212700e+04      1.000000         0.000000       0.000000   
25%    2.852056e+07      1.000000         1.000000       0.990000   
50%    5.673150e+07      2.000000         1.000000       1.610000   
75%    8.537452e+07      2.000000         2.000000       3.060000   
max    1.134863e+08      2.000000         6.000000      33.960000   

         RatecodeID  PULocationID  DOLocationID  payment_type   fare_amount  \
count  22699.000000  22699.000000  22699.000000  22699.000000  22699.000000   
mean       1.043394    162.412353    161.527997      1.336887     13.026629   
std        0.708391     66.633373     70.139691      0.496211     13.243791   
min        1.000000      1.000000      1.000000      1.000000   -120.000000   
25%        1.000000    114.000000    112.000000      1.000000      6.500000   
50%        1.000000    162.000000    162.000000      1.000000      9.500000   
75%        1.000000    233.000000    233.000000      2.000000     14.500000   
max       99.000000    265.000000    265.000000      4.000000    999.990000   

              extra       mta_tax    tip_amount  tolls_amount  \
count  22699.000000  22699.000000  22699.000000  22699.000000   
mean       0.333275      0.497445      1.835781      0.312542   
std        0.463097      0.039465      2.800626      1.399212   
min       -1.000000     -0.500000      0.000000      0.000000   
25%        0.000000      0.500000      0.000000      0.000000   
50%        0.000000      0.500000      1.350000      0.000000   
75%        0.500000      0.500000      2.450000      0.000000   
max        4.500000      0.500000    200.000000     19.100000   

       improvement_surcharge  total_amount  
count           22699.000000  22699.000000  
mean                0.299551     16.310502  
std                 0.015673     16.097295  
min                -0.300000   -120.300000  
25%                 0.300000      8.750000  
50%                 0.300000     11.800000  
75%                 0.300000     17.800000  
max                 0.300000   1200.290000

### **Task 2c. Understand the data - Investigate the variables**

Sort and interpret the data table for two variables:`trip_distance` and `total_amount`.

**Answer the following three questions:**

**Question 1:** Sort your first variable (`trip_distance`) from maximum to minimum value, do the values seem normal?

**Question 2:** Sort by your second variable (`total_amount`), are any values unusual?

**Question 3:** Are the resulting rows similar for both sorts? Why or why not?

Question 1: The max values of trip_distance range from approx. 30 - 34. These trips likely take passengers outside of the city. The min values of trip_distance are 0, which might mean that passengers hailed a cab/ride, got in the vehicle and waited, and then decided not to go anywhere - the fare amounts for those trips might just reflect wait time for the cab/ride. However, some of those trips have different pickup and dropoff locations listed - in those cases, perhaps the tool used to measure distance was inoperable. It would be prudent to investigate these outliers to determine root cause and decide whether to include or exclude these in various analyses.

Question 2: The top two highest total_amount trips are significantly higher than the rest at 1200.29 and 450.30. The next three highest after those are over 200. On the low end of total_amount, there are several negative values, with one value of -120.30 being significantly greater in magnitude than the others. Perhaps these are refunds?

Question 3: The top total_amounts listed in the trip_distance sort are included in the top total_amounts in the total_amount sort but there are several top total_amount values in the total_amount sort not listed in the top values for total_amount in the trip_distance sort. Some of the trip_distance values for the top total_amount values in the total_amount sort are listed at 0. While trip_distance might generally predict total_amount overall, it is clear that it does not do so in all instances (the correlation is not perfect). It would be prudent to explore p-values from multiple regression analysis to identify perhaps a system of variables that would better predict total_amount.

In [4]:
df_sort = df.sort_values(by = 'trip_distance',ascending = False)

df_sort.head(10)

# Sort the data by trip distance from maximum to minimum value

Unnamed: 0  VendorID    tpep_pickup_datetime   tpep_dropoff_datetime  \
9280     51810714         2  06/18/2017 11:33:25 PM  06/19/2017 12:12:38 AM   
13861    40523668         2   05/19/2017 8:20:21 AM   05/19/2017 9:20:30 AM   
6064     49894023         2  06/13/2017 12:30:22 PM   06/13/2017 1:37:51 PM   
10291    76319330         2  09/11/2017 11:41:04 AM  09/11/2017 12:18:58 PM   
29       94052446         2   11/06/2017 8:30:50 PM  11/07/2017 12:00:00 AM   
18130    90375786         1   10/26/2017 2:45:01 PM   10/26/2017 4:12:49 PM   
5792     68023798         2   08/11/2017 2:14:01 PM   08/11/2017 3:17:31 PM   
15350    77309977         2   09/14/2017 1:44:44 PM   09/14/2017 2:34:29 PM   
10302    43431843         1   05/15/2017 8:11:34 AM   05/15/2017 9:03:16 AM   
2592     51094874         2   06/16/2017 6:51:20 PM   06/16/2017 7:41:42 PM   

       passenger_count  trip_distance  RatecodeID store_and_fwd_flag  \
9280                 2          33.96           5                  N   
13861                1          33.92           5                  N   
6064                 1          32.72           3                  N   
10291                1          31.95           4                  N   
29                   1          30.83           1                  N   
18130                1          30.50           1                  N   
5792                 1          30.33           2                  N   
15350                1          28.23           2                  N   
10302                1          28.20           2                  N   
2592                 1          27.97           2                  N   

       PULocationID  DOLocationID  payment_type  fare_amount  extra  mta_tax  \
9280            132           265             2       150.00    0.0      0.0   
13861           229           265             1       200.01    0.0      0.5   
6064            138             1             1       107.00    0.0      0.0   
10291           138           265             2       131.00    0.0      0.5   
29              132            23             1        80.00    0.5      0.5   
18130           132           220             1        90.50    0.0      0.5   
5792            132           158             1        52.00    0.0      0.5   
15350            13           132             1        52.00    0.0      0.5   
10302            90           132             1        52.00    0.0      0.5   
2592            261           132             2        52.00    4.5      0.5   

       tip_amount  tolls_amount  improvement_surcharge  total_amount  
9280         0.00          0.00                    0.3        150.30  
13861       51.64          5.76                    0.3        258.21  
6064        55.50         16.26                    0.3        179.06  
10291        0.00          0.00                    0.3        131.80  
29          18.56         11.52                    0.3        111.38  
18130       19.85          8.16                    0.3        119.31  
5792        14.64          5.76                    0.3         73.20  
15350        4.40          5.76                    0.3         62.96  
10302       11.71          5.76                    0.3         70.27  
2592         0.00          5.76                    0.3         63.06

In [8]:
total_amount_desc_sort = df.sort_values(by = 'total_amount',ascending = False)

print(total_amount_desc_sort)

print(total_amount_desc_sort.head(20)['total_amount'])

# Sort the data by total amount and print the top 20 values

       Unnamed: 0  VendorID    tpep_pickup_datetime   tpep_dropoff_datetime  \
8476     11157412         1   02/06/2017 5:50:10 AM   02/06/2017 5:51:08 AM   
20312   107558404         2   12/19/2017 9:40:46 AM   12/19/2017 9:40:55 AM   
13861    40523668         2   05/19/2017 8:20:21 AM   05/19/2017 9:20:30 AM   
12511   107108848         2   12/17/2017 6:24:24 PM   12/17/2017 6:24:42 PM   
15474    55538852         2   06/06/2017 8:55:01 PM   06/06/2017 8:55:06 PM   
...           ...       ...                     ...                     ...   
11204    58395501         2   07/09/2017 7:20:59 AM   07/09/2017 7:23:50 AM   
14714   109276092         2  12/24/2017 10:37:58 PM  12/24/2017 10:41:08 PM   
17602    24690146         2   03/24/2017 7:31:13 PM   03/24/2017 7:34:49 PM   
20698    14668209         2  02/24/2017 12:38:17 AM  02/24/2017 12:42:05 AM   
12944    29059760         2  04/08/2017 12:00:16 AM  04/08/2017 11:15:57 PM   

       passenger_count  trip_distance  RatecodeID s

In [9]:
total_amount_asc_sort = df.sort_values(by = 'total_amount',ascending = True)

print(total_amount_asc_sort)

print(total_amount_asc_sort.head(20)['total_amount'])

# Sort the data by total amount and print the bottom 20 values

       Unnamed: 0  VendorID    tpep_pickup_datetime   tpep_dropoff_datetime  \
12944    29059760         2  04/08/2017 12:00:16 AM  04/08/2017 11:15:57 PM   
20698    14668209         2  02/24/2017 12:38:17 AM  02/24/2017 12:42:05 AM   
17602    24690146         2   03/24/2017 7:31:13 PM   03/24/2017 7:34:49 PM   
11204    58395501         2   07/09/2017 7:20:59 AM   07/09/2017 7:23:50 AM   
14714   109276092         2  12/24/2017 10:37:58 PM  12/24/2017 10:41:08 PM   
...           ...       ...                     ...                     ...   
15474    55538852         2   06/06/2017 8:55:01 PM   06/06/2017 8:55:06 PM   
12511   107108848         2   12/17/2017 6:24:24 PM   12/17/2017 6:24:42 PM   
13861    40523668         2   05/19/2017 8:20:21 AM   05/19/2017 9:20:30 AM   
20312   107558404         2   12/19/2017 9:40:46 AM   12/19/2017 9:40:55 AM   
8476     11157412         1   02/06/2017 5:50:10 AM   02/06/2017 5:51:08 AM   

       passenger_count  trip_distance  RatecodeID s

In [17]:
df['payment_type'].value_counts()

# How many of each payment type are represented in the data?

1    15265
2     7267
3      121
4       46
Name: payment_type, dtype: int64

According to the data dictionary, the payment method was encoded as follows:

1 = Credit card  
2 = Cash  
3 = No charge  
4 = Dispute  
5 = Unknown  
6 = Voided trip

In [21]:
credit_card_mask = df['payment_type'] == 1

print(df[credit_card_mask]['tip_amount'].mean())

# What is the average tip for trips paid for with credit card?

cash_mask = df['payment_type'] == 2

print(df[cash_mask]['tip_amount'].mean())

# What is the average tip for trips paid for with cash?

2.7298001965279934
0.0


In [22]:
df['VendorID'].value_counts()

# How many times is each vendor ID represented in the data?

2    12626
1    10073
Name: VendorID, dtype: int64

In [23]:
vendor_1_mask = df['VendorID'] == 1

print(df[vendor_1_mask]['total_amount'].mean())


vendor_2_mask = df['VendorID'] == 2

print(df[vendor_2_mask]['total_amount'].mean())

# What is the mean total amount for each vendor?

16.298118733246966
16.32038175193886


In [3]:
credit_card_mask = df['payment_type'] == 1

print(df[credit_card_mask])

# Filter the data for credit card payments only

print(df[credit_card_mask]['passenger_count'])

print(df[credit_card_mask]['passenger_count'].value_counts())

# Filter the credit-card-only data for passenger count only

       Unnamed: 0  VendorID    tpep_pickup_datetime   tpep_dropoff_datetime  \
0        24870114         2   03/25/2017 8:55:43 AM   03/25/2017 9:09:47 AM   
1        35634249         1   04/11/2017 2:53:28 PM   04/11/2017 3:19:58 PM   
2       106203690         1   12/15/2017 7:26:56 AM   12/15/2017 7:34:08 AM   
3        38942136         2   05/07/2017 1:17:59 PM   05/07/2017 1:48:14 PM   
5        23345809         2   03/25/2017 8:34:11 PM   03/25/2017 8:42:11 PM   
...           ...       ...                     ...                     ...   
22692    60425673         1   07/16/2017 3:22:51 AM   07/16/2017 3:40:52 AM   
22693    67858616         2  08/10/2017 10:20:04 PM  08/10/2017 10:29:31 PM   
22695    66632549         2   08/06/2017 4:43:59 PM   08/06/2017 5:24:47 PM   
22697    60217333         2  07/15/2017 12:56:30 PM   07/15/2017 1:08:26 PM   
22698    17208911         1   03/02/2017 1:02:49 PM   03/02/2017 1:16:09 PM   

       passenger_count  trip_distance  RatecodeID s

In [29]:
credit_card_mask = df['payment_type'] == 1

print(df[credit_card_mask].groupby('passenger_count').agg({'tip_amount': 'mean'}))

# Calculate the average tip amount for each passenger count (credit card payments only)


                 tip_amount
passenger_count            
0                  2.610370
1                  2.714681
2                  2.829949
3                  2.726800
4                  2.607753
5                  2.762645
6                  2.643326


<img src="images/Construct.png" width="100" height="100" align=left>

## PACE: **Construct**

**Note**: The Construct stage does not apply to this workflow. The PACE framework can be adapted to fit the specific requirements of any project. 




<img src="images/Execute.png" width="100" height="100" align=left>

## PACE: **Execute**

Consider the questions in your PACE Strategy Document and those below to craft your response.


### **Given your efforts, what can you summarize for DeShawn and the data team?**

*Note for Learners: Your notebook should contain data that can address Luana's requests. Which two variables are most helpful for building a predictive model for the client: NYC TLC?*

DeShawn, Luana, and team,

I've imported and conducted an inital inspection of the 2017 Yellow Taxi Trip Data dataset and have the following observations/recommendations.

-NON-NULL and NULL VALUES. There are no null values in the dataset. While the dataset is supposed to have 408,294 rows, when I imported it it only shows as having 22,699 rows. We should confirm with TLC whether or not this dataset is complete, and, if not, get the complete dataset.

-KEY VARIABLES: On the surface, it seems that the most important variables to start with for including in a regression model that predicts trip fares might be trip distance and trip duration. We'd need to create a new variable for trip duration by taking the difference between the "tpep_dropoff_datetime" and "tpep_pickup_datetime" columns. For trip duration, it might be helpful to calculate average trip durations between each different pickup and dropoff location during both peak and non-peak hours. Of course, we'll need to test this hypothesis by determining significance level of p-values and adjusted R-squared when building/testing the model.

-COLUMN DTYPES. The dataset has 18 columns with the following labels and dtypes:

"Unnamed: 0": int64
"VendorID": int64
"tpep_pickup_datetime": object (date-time groups in the format of "MM/DD/YYYY HH:MM:SS AM/PM")
"tpep_dropoff_datetime": object (date-time groups in the format of "MM/DD/YYYY HH:MM:SS AM/PM")
"passenger_count": int64
"trip_distance": float64
"RatecodeID": int64
"store_and_fwd_flag": object (boolean Y or N)
"PULocationID": int64
"DOLocationID": int64
"payment_type": int64
"fare_amount": float64
"extra": float64
"mta_tax": float64
"tip_amount": float64
"tolls_amount": float64
"improvement_surcharge": float64
"total_amount": float64


ADDITIONAL OBESERVATIONS/RECOMMENDATIONS


-The "Unnamed: 0" column appears to contain unique IDs for each trip. We should confirm this with TLC.

-There are many trips with 0 distance reported even though they have different pickup and dropoff location IDs. We should get with TLC to better understand these and other similar cases so we can deal with them appropriately when conducting analysis.

-Similarly, there are trips with 0 passenger counts reported. We should ask TLC about these as well to gain better understanding.

-We should create a new variable for total_amount that doesn't include tip amounts since tips are not required for trips. Using this new variable as the dependent variable in the regression model will help ensure our model isn't skewed.

-Lastly, according to the dataset's metadata (https://data.cityofnewyork.us/Transportation/2017-Yellow-Taxi-Trip-Data/biws-g3hs), this dataset was provided to TLC by a tech company and TLC cannot confirm its accuracy. It would be prudent to contact this tech company to confirm the dataset's accuracy and to determine how the data was gathered and combined for all variables.


Please let me know if you have any questions/further guidance. I'm really looking forward to working with you and the NYC TLC to complete this project!

Respectfully,

Chris Aguirre
Data Analyst
Automatidata

**Congratulations!** You've completed this lab. However, you may not notice a green check mark next to this item on Coursera's platform. Please continue your progress regardless of the check mark. Just click on the "save" icon at the top of this notebook to ensure your work has been logged.